docs/extractor.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Use a for-loop, to access
+# multiple containers. The command-line argument
+# is a release value, like "22.02".
+# Between the release value and the container name,
+# there is enough information to group the container
+# information...
+#
+# {
+#    "merlin-training": {
+#       "22.02": {
+#          "cuda": "11.6"
+#       },
+#       "22.01": {
+#          "cuda": "11.5"
+#       }
+#    }
+# }
+#
+# After all the data is gathered, it should be possible
+# to construct a multi-release table.
+# "merlin-training"
+# |       | 22.02 | 22.01 |
+# | ----- | ----- | ----- |
+# | CUDA  |  11.6 |  11.5 |
+
+import argparse
+import contextlib
+import json
+import logging
+import os
+from collections import defaultdict
+from datetime import date
+from pathlib import Path
+
+import yaml
+from mergedeep import merge
+
+import docker
+
+level = logging.DEBUG if os.environ.get("DEBUG") else logging.INFO
+logging.basicConfig(level=level)
+logger = logging.getLogger("extractor")
+
+
+@contextlib.contextmanager
+def managed_container(img):
+    client = docker.from_env()
+    container = None
+    try:
+        # runtime="nvidia",
```

Comment:
The runtime isn't installed on the GitHub runner.  IIRC (and that's dicey) it produced an error on my first run.  I don't need it for the data extraction because I'm not running any analytics.

Reply:
That makes sense!